### DISCUSSION TITLE
Multi-processing with IterableDataset Warning

### DISCUSSION THREAD

**Discussion by ozen** [Lightning-AI/pytorch-lightning#6940] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello. The warning message given when using an IterableDataset that has `__len__` defined got me confused. 

https://github.com/PyTorchLightning/pytorch-lightning/blob/049006a59cf62d3539aaf19037aca1caac98abfd/pytorch_lightning/utilities/data.py#L40-L45

Is the part `e.g. batch size > 1` should be `e.g. num_workers > 1` or am I missing something?

Is this warning about what [IterableDataset doc](https://pytorch.org/docs/stable/data.html#torch.utils.data.IterableDataset) explains about configuring the workers? If I write my IterableDataset similarly to the example provided in the doc and also define a `__len__` function, will there be any problems due to using PL?

Thanks.

**Comment by awaelchli** [comment]
It was fixed in #6972, the message should be clearer now.
A longer explanation from pytorch can be found [here](https://github.com/pytorch/pytorch/blob/399b66c8137db9ef86e45707ec30a44e40495f62/torch/utils/data/dataloader.py#L374).
